### PR TITLE
fix pattern lookup match escaping on F expressions

### PR DESCRIPTION
**Pull request by timgraham**
Escape $regexMatch special characters instead of LIKE special chars.

This is complemented by a patch to the Django fork which fails unless each of these replacements are made:
```diff
diff --git a/tests/expressions/tests.py b/tests/expressions/tests.py
index 36624649e5..60c88e42e2 100644
--- a/tests/expressions/tests.py
+++ b/tests/expressions/tests.py
@@ -1229,6 +1229,17 @@ class ExpressionsTests(TestCase):
                 Employee(firstname="Jean-Claude", lastname="Claude%"),
                 Employee(firstname="Johnny", lastname="Joh\\n"),
                 Employee(firstname="Johnny", lastname="_ohn"),
+                Employee(firstname="Johnny", lastname="^Joh"),
+                Employee(firstname="Johnny", lastname="Johnny$"),
+                Employee(firstname="Johnny", lastname="Joh."),
+                Employee(firstname="Johnny", lastname="[J]ohnny"),
+                Employee(firstname="Johnny", lastname="(J)ohnny"),
+                Employee(firstname="Johnny", lastname="J*ohnny"),
+                Employee(firstname="Johnny", lastname="J+ohnny"),
+                Employee(firstname="Johnny", lastname="J?ohnny"),
+                Employee(firstname="Johnny", lastname="J{1}ohnny"),
+                Employee(firstname="Johnny", lastname="J|ohnny"),
+                Employee(firstname="Johnny", lastname="J-ohnny"),
             ]
         )
         claude = Employee.objects.create(firstname="Jean-Claude", lastname="Claude")
```

### CODE REVIEW

**Comment by WaVEV** [comment]
Is it ready to review?

**Review by WaVEV — APPROVED** [review]

File: django_mongodb/lookups.py [hunk 1]
```diff
@@ -24,11 +24,28 @@ def is_null(self, compiler, connection):
...
+    (".", r"\."),  # match any character
+    ("[", r"\["),  # start character class definition
+    ("|", r"\|"),  # start of alternative branch
+    ("(", r"\("),  # start group or control verb
+    (")", r"\)"),  # end group or control verb
+    ("*", r"\*"),  #  0 or more quantifier
+    ("+", r"\+"),  #  1 or more quantifier
+    ("?", r"\?"),  # 0 or 1 quantifier
+    ("{", r"\}"),  # start min/max quantifier
+)
```
Comment (Jibola) [inline django_mongodb/lookups.py:41]: So this will trigger 13 `$replaceAll` when doing lookups?

Comment (WaVEV) [reply django_mongodb/lookups.py:41]: is there a way to do that replacements in python? and pass the final string

Comment (timgraham) [reply django_mongodb/lookups.py:41]: When the value to be matched is a string, `re.escape()` [handles it](https://github.com/mongodb-labs/django-mongodb/blob/5f2a094293636f6d624f4b258f155c8fd01b46d2/django_mongodb/operations.py#L111-L113). This code is only for the case when `regexMatch`'s `regex` is a column.

**Review by Jibola — APPROVED** [review]

**Pull request merged**